### PR TITLE
Add efficient methods for taking / transferring geometries from collections

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: src/core/geometry/qgsgeometrycollection.h [hunk 1]
```diff
@@ -204,6 +204,15 @@ class CORE_EXPORT QgsGeometryCollection: public QgsAbstractGeometry
     //! Adds a geometry and takes ownership. Returns TRUE in case of success.
     virtual bool addGeometry( QgsAbstractGeometry *g SIP_TRANSFER );
 
+    /**
+     * Adds a list of geometries to the collection, transferring ownership to the collection.
+     *
+     * Returns TRUE in case of success.
```
Comment (nirvn) [inline src/core/geometry/qgsgeometrycollection.h:210]: \returns instead?

**Review by nirvn — APPROVED** [review]

File: src/core/geometry/qgsgeometrycollection.h (hunk 1 above)
Comment (nyalldawson) [reply src/core/geometry/qgsgeometrycollection.h:210]: image

:rofl:

Comment (nyalldawson) [reply src/core/geometry/qgsgeometrycollection.h:210]: (I was going to suggest we address that in bulk in a follow up, but I'm inclined not to if there's that many hits!)

Comment (nirvn) [reply src/core/geometry/qgsgeometrycollection.h:210]: :scream: 

Alright, it's something that should be addressed (if it is to be changed) in a different PR.

**Pull request merged**